packages/table/components/Table.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from "react";
+import { css } from "emotion";
+import { AutoSizer, MultiGrid } from "react-virtualized";
+import { IColumnProps, Column } from "./Column";
+import memoizeOne from "memoize-one";
+import { coreColors as Colors } from "../../shared/styles/color";
+
+export interface ITableProps {
+  fixedColumnCount: number;
+  enableFixedColumnScroll: boolean;
+  enableFixedRowScroll: boolean;
+  data: any;
+  children:
+    | Array<React.ReactElement<IColumnProps>>
+    | React.ReactElement<IColumnProps>;
+}
+
+export class Table extends React.PureComponent<ITableProps, {}> {
+  public multiGridRef: { recomputeGridSize?: any } = {};
+
+  public getData = memoizeOne(
+    (data: any[]): any[] => {
+      return [{}, ...data];
+    }
+  );
+
+  public getColumnSizes = memoizeOne(
+    (
+      children: Array<React.ReactElement<IColumnProps>>,
+      width: number
+    ): number[] => {
+      const totalColumns: number = children.length;
+      let remainingWidth: number = width;
+      return children.map((child, currentIndex) => {
+        const calculatedWidth = child.props.width({
+          width,
+          totalColumns,
+          currentIndex,
+          remainingWidth
+        });
+        remainingWidth -= calculatedWidth;
+        return calculatedWidth;
+      });
+    }
+  );
+
+  constructor(props) {
+    super(props);
+
+    this.getCell = this.getCell.bind(this);
+    this.setRef = this.setRef.bind(this);
+    this.updateGrid = this.updateGrid.bind(this);
+  }
+
+  public getCell({ key, columnIndex, rowIndex, style }) {
+    const data = this.getData(this.props.data);
+
+    const column = React.Children.toArray(this.props.children)[
+      columnIndex
+    ] as React.ReactElement<IColumnProps>;
+
+    if (!column || column.type !== Column) {
+      return null;
+    }
+
+    let className = css({
+      borderBottom: `1px solid ${Colors().greyLight}`,
+      boxSizing: "border-box",
+      ...style
+    });
+
+    if (rowIndex === 0) {
+      className = css({
+        boxSizing: "border-box",
+        borderBottom: `1px solid ${Colors().black}`,
+        ...style
+      });
+      return <div {...{ className, key }}>{column.props.label}</div>;
+    }
+
+    return (
+      <div {...{ className, key }}>
+        {column.props.cellRenderer(data[rowIndex])}
+      </div>
+    );
+  }
+
+  public updateGrid() {
+    if (this.multiGridRef.recomputeGridSize != null) {
+      this.multiGridRef.recomputeGridSize();
+    }
+  }
+
+  public setRef(ref: any) {
+    this.multiGridRef = ref;
+  }
+
+  public render() {
+    const {
+      fixedColumnCount,
+      enableFixedColumnScroll,
+      enableFixedRowScroll
+    } = this.props;
+    const mergedData = this.getData(this.props.data);
+    const columnCount = React.Children.count(this.props.children);
+
+    return (
+      <AutoSizer onResize={this.updateGrid}>
```

Comment:
Update grid made me for a second a bit worried that we update values in the grid or something the like. Can we use a different name, maybe updateGridSize?

Reply:
Yes, that makes complete sense sorry to make you feel that way.

Reply:
done